src/mainview.tsx

```diff
@@ -764,6 +772,35 @@ export class MainView extends React.Component<IProps, IStates> {
     });
   };
 
+  private _onSharedModelChanged(
+    sender: JupyterCadModel,
+    change: IJupyterCadDocChange
+  ): void {
+    if (change.objectChange) {
+      let visible = false;
+
+      change.objectChange.forEach(change => {
+        if (change.key === 'visible') {
+          visible = true;
+          const obj = this._meshGroup?.getObjectByName(change.name);
+          if (obj) {
+            obj.visible = change.newValue?.visible ?? false;
+          }
+        }
+      });
+
+      if (!visible) {
+        this._postMessage({
+          action: WorkerAction.LOAD_FILE,
+          payload: {
+            fileName: this._context.path,
+            content: this._model.getContent()
+          }
+        });
+      }
```

Comment:
I'm not sure I understand this part. What is it doing?

Reply:
I think the variable `visible` should be named `visibleChanged`. 
But I'm wondering what is the behavior of this function if I have both `visible` and other keys in the `change.objectChange` list. We should send the message to the worker to re-render the scene in this case.

Reply:
I'm not sure it matters much. I'm currently working on the guidata options and having the visible property there. So this method will actually be triggered on an `sharedOptionChange` event, not an object change.

Reply:
We have both `visible` properties in the document and the GUI document. When we click on the visibility icon in the tree view, which `visible` property we should update? If it's both, then the viewer will get both `sharedOptionChange` and `sharedObjectChange`.

Reply:
I believe it should be the `visible` property from the GUI data that we should change.

Reply:
I don't know when we would change the visible property from the object data though. I don't understand why freecad has this separation.

Also, we should probably give the possibility for the user to not sync the guidata state.

Reply:
> I believe it should be the `visible` property from the GUI data that we should change.

Me too! The script you used can update the visibility of objects in the GUI document?

Reply:
Yes it can :)

Reply:
So let's get https://github.com/QuantStack/jupytercad/pull/91 in first then we will adapt this PR?

Reply:
I'd prefer merging #83 first actually, because I already rebased my work on this locally

Reply:
ok, I will try to fix the integration tests before merging this PR.

Reply:
> But I'm wondering what is the behavior of this function if I have both visible and other keys in the change.objectChange list. We should send the message to the worker to re-render the scene in this case.

When changing the visibility of the object from the tree view the transaction has only one change, the `visible` key. I think is not possible to have an event that changes multiple keys when changing the visibility.